src/structures/ApplicationCommand.ts

```diff
@@ -0,0 +1,39 @@
+import BulbBotClient from "./BulbBotClient";
+import { CommandInteraction, Permissions, PermissionString } from "discord.js";
+import { ApplicationCommandOptions, ApplicationCommandType } from "../utils/types/ApplicationCommands";
+
+export default class ApplicationCommand {
+	public readonly client: BulbBotClient;
+	public readonly type: ApplicationCommandType;
+	public readonly name: string;
+	public readonly description: string;
+	public readonly dm_permissions: boolean;
+	public readonly default_member_permissions: string | null;
+	public readonly command_permissions: PermissionString[];
+	public readonly options: ApplicationCommandOptions[];
+
+	constructor(client: BulbBotClient, options: any) {
+		this.client = client;
+		this.type = options.type;
+		this.name = options.name;
+		this.description = options.description;
+		this.dm_permissions = options.dm_permissions || false;
+		this.command_permissions = options.command_permissions || [];
+		this.default_member_permissions = this._computePermissions();
```

Comment:
Don't we typically use camelCase, not snake_case? We should maintain consistency. I think there are lint rules we can turn on to enforce that, if they aren't already

Reply:
This can be enforced here as this is not the part we are sending to the discord api to register the slash commands

Reply:
What gets sent to the API for slash commands? I assume that part requires snake_case?

Reply:
The API does require snake_case yeah. Could be changed to maintain consistency for sure